src/file-selector.ts

```diff
@@ -105,23 +105,29 @@ function flatten<T>(items: any[]): T[] {
   );
 }
 
-function fromDataTransferItem(
+async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  if (typeof (item as any).getAsFileSystemHandle === "function") {
-    return (item as any).getAsFileSystemHandle().then(async (h: any) => {
-      const file = await h.getFile();
-      file.handle = h;
-      return toFileWithPath(file);
-    });
+): Promise<FileWithPath> {
+  if (typeof item.getAsFileSystemHandle === "function") {
+    const handle = await item.getAsFileSystemHandle();
+
+    if (!handle || !isFileHandle(handle)) {
+      throw new Error("Transferred item is not a file.");
+    }
+
+    const file = await handle.getFile();
+    (file as any).handle = handle;
+    return toFileWithPath(file);
   }
+
   const file = item.getAsFile();
+
   if (!file) {
-    return Promise.reject(`${item} is not a File`);
```

Comment:
The intent was to see what exactly is not a file ... so it might make sense to make a custom error object that can embed the `item` which could later be inspected if they look at the error?

Reply:
We could do that yeah, proper custom error types [are a bit involved](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#custom_error_types) though, so I'd rather avoid it. I also don't know how much value there would be in having the `item` available here.